### PR TITLE
Wip/hughsie/ccgx2

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Comment by hughsie** [comment]
@chlee75

**Comment by chlee75** [comment]
I have tested segmentation fault with this patch.
There is no error as far. I think segmentation fault has been fixed.
Could you please upload it to LVFS?
Ryan.

File: plugins/ccgx/fu-ccgx-hid-device.c [hunk 1]
```diff
@@ -44,13 +45,47 @@ fu_ccgx_hid_device_detach (FuDevice *device, GError **error)
...
+	g_autoptr(GError) error = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* do this in idle so that the original HID device gets added */
+	locker = fu_device_locker_new (self, &error);
+	if (locker == NULL) {
+		g_warning ("failed to open HID device: %s",
+			   error->message);
+	} else {
+		if (!fu_ccgx_hid_device_detach (FU_DEVICE (self), &error)) {
```
Comment (superm1) [inline plugins/ccgx/fu-ccgx-hid-device.c]: What actually happens?  Does the FuDevice disappear?  I worry this is a timing hack that might not behave the same reliably.

Comment (hughsie) [reply plugins/ccgx/fu-ccgx-hid-device.c]: Yes, it unplugs and replugs with a different VID PID.

Comment (hughsie) [reply plugins/ccgx/fu-ccgx-hid-device.c]: actually, maybe a cleaner thing to do might be to detach and then return FALSE...

Comment (hughsie) [reply plugins/ccgx/fu-ccgx-hid-device.c]: actually 2, we need to add the device sometimes as in future hardware the HID mode is updatable, so we can't return failure from setup.

Comment (superm1) [reply plugins/ccgx/fu-ccgx-hid-device.c]: Well you could also just have a dummy setup method and do the detach in probe perhaps?

Comment (hughsie) [reply plugins/ccgx/fu-ccgx-hid-device.c]: I've switched it to returning FALSE from setup. Setup already has the device open so it's the natural time to do it. If/when we have a device that can be flashing in HID mode we can just do a different thing in setup.

**Comment by hughsie** [comment]
@chlee75 can you verify this fixed version still works please?

**Comment by hughsie** [comment]
> this fixed version still works please

It just worked for me using the embargo remote.

**Pull request merged**